cmsdb/campaigns/run3_2022_preEE_nano_v11/top.py

```diff
@@ -0,0 +1,553 @@
+
+# dileptonic decay (top)
+# GrASP: https://cms-pdmv.cern.ch/grasp/samples?campaign=Run3Summer22wmLHEGS&nanoaod_version=v11&dataset=TWminusto2L2Nu*_TuneCP5%28%7CUp%7CDown%7CCR1%7CCR2%29_13p6TeV&chained_request=Premix
+cpn.add_dataset(
+    name="st_twchannel_t_dl_powheg",
+    id=14678180,
+    is_data=False,
+    processes=[procs.st_twchannel_t_dl],
+    info=dict(
+        nominal=DatasetInfo(
+            keys=[
+                "/TWminusto2L2Nu_TuneCP5_13p6TeV_powheg-pythia8/Run3Summer22NanoAODv11-126X_mcRun3_2022_realistic_v2-v1/NANOAODSIM",
+            ],
+            n_files=4,
+            n_events=2435564,
+        ),
+        tune_up=DatasetInfo(
+            keys=[
+                "/TWminusto2L2Nu_TuneCP5Up_13p6TeV_powheg-pythia8/Run3Summer22NanoAODv11-126X_mcRun3_2022_realistic_v2-v1/NANOAODSIM",
+            ],
+            n_files=4,
+            n_events=2375416,
+        ),
+        tune_down=DatasetInfo(
+            keys=[
+                "/TWminusto2L2Nu_TuneCP5Down_13p6TeV_powheg-pythia8/Run3Summer22NanoAODv11-126X_mcRun3_2022_realistic_v2-v1/NANOAODSIM",
+            ],
+            n_files=4,
+            n_events=2385737,
+        ),
+        cr_1=DatasetInfo(
+            keys=[
+                "/TWminusto2L2Nu_TuneCP5CR1_13p6TeV_powheg-pythia8/Run3Summer22NanoAODv11-126X_mcRun3_2022_realistic_v2-v1/NANOAODSIM",
+            ],
+            n_files=4,
+            n_events=2492355,
+        ),
+        cr_2=DatasetInfo(
+            keys=[
+                "/TWminusto2L2Nu_TuneCP5CR2_13p6TeV_powheg-pythia8/Run3Summer22NanoAODv11-126X_mcRun3_2022_realistic_v2-v1/NANOAODSIM",
+            ],
+            n_files=3,
+            n_events=2497224,
+        ),
+    ),
+)
+
+# dileptonic decay (anti-top)
+# GrASP: https://cms-pdmv.cern.ch/grasp/samples?campaign=Run3Summer22wmLHEGS&nanoaod_version=v11&dataset=TbarWplusto2L2Nu*_TuneCP5%28%7CUp%7CDown%7CCR1%7CCR2%29_13p6TeV&chained_request=Premix
+cpn.add_dataset(
+    name="st_twchannel_tbar_dl_powheg",
+    id=14690995,
+    is_data=False,
+    processes=[procs.st_twchannel_tbar_dl],
+    info=dict(
+        nominal=DatasetInfo(
+            keys=[
+                "/TbarWplusto2L2Nu_TuneCP5_13p6TeV_powheg-pythia8/Run3Summer22NanoAODv11-126X_mcRun3_2022_realistic_v2-v1/NANOAODSIM",
+            ],
+            n_files=3,
+            n_events=2401536,
+        ),
+        # missing as of 2023-07-01
+        # tune_up=DatasetInfo(
+        #     keys=[
+        #         "/TbarWplusto2L2Nu_TuneCP5Up_13p6TeV_powheg-pythia8/Run3Summer22NanoAODv11-126X_mcRun3_2022_realistic_v2-v1/NANOAODSIM",
+        #     ],
+        #     n_files=None,
+        #     n_events=None,
+        # ),
+        tune_down=DatasetInfo(
+            keys=[
+                "/TbarWplusto2L2Nu_TuneCP5Down_13p6TeV_powheg-pythia8/Run3Summer22NanoAODv11-126X_mcRun3_2022_realistic_v2-v1/NANOAODSIM",
+            ],
+            n_files=2,
+            n_events=2422160,
+        ),
+        cr_1=DatasetInfo(
```

Comment:
The CR1/2 postfix refers to a different generator tune, right?
Are CR1 and CR2 somehow related to each other?

Usually, the varied `DatasetInfo` objects in a `Dataset` refer to systematic variations with `_up` and `_down` variations, so we need to figure out how to treat other cases.

(this comment also applies to all other varied datasets)

Reply:
To my knowledge, the CR1 and CR2 refer to alternate color reconnection models in the UE tune. In some sense they can be considered systematic variations, but there is no clear up or down, since completely different models are being used.

According to the MC configuration fragments in CMSSW for the [nominal](https://github.com/cms-sw/cmssw/blob/CMSSW_12_6_0_patch1/Configuration/Generator/python/MCTunesRun3ECM13p6TeV/PythiaCP5Settings_cfi.py), [CR1](https://github.com/cms-sw/cmssw/blob/CMSSW_12_6_0_patch1/Configuration/Generator/python/MCTunesRun3ECM13p6TeV/PythiaCP5CR1TuneSettings_cfi.py) and  [CR2](https://github.com/cms-sw/cmssw/blob/CMSSW_12_6_0_patch1/Configuration/Generator/python/MCTunesRun3ECM13p6TeV/PythiaCP5CR2TuneSettings_cfi.py) samples, they should correspond to setting `ColourReconnection:mode` in Pythia to 0, 1, and 2, respectively. These are explained in the [Pythia manual](https://pythia.org/latest-manual/ColourReconnection.html) as follows:
```
mode  ColourReconnection:mode   (default = 0; minimum = 0; maximum = 4)
Determine which model is used for colour reconnection. Beware that some settings may need to be changed to match the model selected.
option 0 : The MPI-based original Pythia 8 scheme.
option 1 : The new more QCD based scheme. Should be combined ewith BeamRemnants:remnantMode = 1.
option 2 : The new gluon-move model. 
```

While I am not aware of the alternative CR models being used by any of our analyses, I think these might be interesting to study at some point in the future. That being said, I agree we need to eventually figure out if/how we want to use these samples, and to what extent it makes sense to have these as systematic shifts. For now I would keep the corresponding `DatasetInfo` entries as they are.

Reply:
Thanks again for the info!

I had in mind that the dataset info keys should follow the format given by `Shift`'s, but that's actually not true (I also [changed internal argument names in order](https://github.com/riga/order/commit/2af09bcdc67d0d4daecf6f9d8ff5c021ed2f90a8) to reflect that better), so I fully agree to keep the info entries as they are :+1: